### PR TITLE
proposal for adding python-typings to sync api

### PR DESCRIPTION
**Pull request by sourabhsinghs**

### CODE REVIEW

File: L119-python-add-typing-to-sync-api.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+Add Typing to gRPC Python Sync API
+----
+* Author(s): Sourabh Singh
+* Approver: gnossen
+* Status: Draft
+* Implemented in: Python
+* Last updated: 22/08/2024
+* Discussion at: <google group thread> (filled after thread exists)
```
[the remark this answers is not in the document]
Comment (sourabhsinghs) [reply L119-python-add-typing-to-sync-api.md]: added google group link thread

File: L119-python-add-typing-to-sync-api.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+Add Typing to gRPC Python Sync API
+----
+* Author(s): Sourabh Singh
+* Approver: gnossen
+* Status: Draft
+* Implemented in: Python
+* Last updated: 22/08/2024
```
[the remark this answers is not in the document]
Comment (sourabhsinghs) [reply L119-python-add-typing-to-sync-api.md]: updated date format

File: L119-python-add-typing-to-sync-api.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+def __call__(self,
+             request: RequestType,
+             timeout: Optional[float] = None,
+             metadata: Optional[MetadataType] = None,
+             credentials: Optional[CallCredentials] = None,
+             wait_for_ready: Optional[bool] = None,
+             compression: Optional[Compression] = None) -> ResponseType:
+```
+
+## Implementation
```
Comment (XuanWang-Amos) [inline L119-python-add-typing-to-sync-api.md:81]: I think we can mention that we're also adding `mypy` and runtime type checker `typeguard` to our CI.

Comment (sourabhsinghs) [reply L119-python-add-typing-to-sync-api.md:81]: updated point to mention mypy and typeguard

**Review by XuanWang-Amos — APPROVED** [review]

File: L119-python-add-typing-to-sync-api.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
+Add Typing to gRPC Python Sync API
+----
+* Author(s): Sourabh Singh
+* Approver: gnossen
+* Status: Draft
+* Implemented in: Python
+* Last updated: 2024/08/23
```
Comment (XuanWang-Amos) [inline L119-python-add-typing-to-sync-api.md:7]: Please update this date before merging.

**Pull request merged**